server/app/com/fortysevendeg/exercises/Actions.scala

```diff
@@ -13,8 +13,19 @@ import play.api.Logger
 
 case class Secure[A](action: Action[A]) extends Action[A] {
   def apply(request: Request[A]): Future[Result] = {
-    if (Play.isProd && !request.secure) {
-      val secureUrl = s"https://${request.domain}${request.uri}"
+
+    val inWWW = request.domain.startsWith("www.")
+
+    val redirect =
```

Comment:
this can be `Play.isProd && (!request.secure || !inWWW)` but not sure if it's more clear.